src/material-experimental/mdc-progress-spinner/progress-spinner.ts

```diff
@@ -0,0 +1,268 @@
+    addClass: (className: string) => this._rootElement.classList.add(className),
+    hasClass: (className: string) => this._rootElement.classList.contains(className),
+    removeClass: (className: string) => this._rootElement.classList.remove(className),
+    removeAttribute: (name: string) => this._rootElement.removeAttribute(name),
+    setAttribute: (name: string, value: string) => this._rootElement.setAttribute(name, value),
+    getDeterminateCircleAttribute: (attributeName: string) => this._determinateCircle.getAttribute(
+      attributeName),
+    setDeterminateCircleAttribute: (attributeName: string,
+                                    value: string) => this._determinateCircle.setAttribute(
+      attributeName,
+      value),
+  };
+
+  constructor(public _elementRef: ElementRef<HTMLElement>,
+              @Optional() @Inject(ANIMATION_MODULE_TYPE) animationMode: string,
+              @Inject(MAT_PROGRESS_SPINNER_DEFAULT_OPTIONS)
+                defaults?: MatProgressSpinnerDefaultOptions) {
+    super(_elementRef);
+    this._noopAnimations = animationMode === 'NoopAnimations' &&
+      (!!defaults && !defaults._forceAnimations);
+
+    if (defaults) {
+      if (defaults.diameter) {
+        this.diameter = defaults.diameter;
+      }
+
+      if (defaults.strokeWidth) {
+        this.strokeWidth = defaults.strokeWidth;
+      }
+    }
+  }
+
+  private _mode: ProgressSpinnerMode = 'determinate';
+
+  /**
+   * Mode of the progress bar.
+   *
+   * Input must be one of these values: determinate, indeterminate, buffer, query, defaults to
+   * 'determinate'.
+   * Mirrored to mode attribute.
+   */
+  @Input()
+  get mode(): ProgressSpinnerMode {
+    return this._mode;
+  }
+
+  set mode(value: ProgressSpinnerMode) {
+    // Note that we don't technically need a getter and a setter here,
+    // but we use it to match the behavior of the existing mat-progress-bar.
+    this._mode = value;
+    this._syncFoundation();
+  }
+
+  private _value = 0;
+
+  /** Value of the progress bar. Defaults to zero. Mirrored to aria-valuenow. */
+  @Input()
+  get value(): number {
+    return this.mode === 'determinate' ? this._value : 0;
+  }
+
+  set value(v: number) {
+    this._value = Math.max(0, Math.min(100, coerceNumberProperty(v)));
+    this._syncFoundation();
+  }
+
+  private _diameter = BASE_SIZE;
+
+  /** The diameter of the progress spinner (will set width and height of svg). */
+  @Input()
+  get diameter(): number {
+    return this._diameter;
+  }
+
+  set diameter(size: number) {
+    this._diameter = size;
+    this._syncFoundation();
+  }
+
+  private _strokeWidth: number;
+
+  /** Stroke width of the progress spinner. */
+  @Input()
+  get strokeWidth(): number {
+    return this._strokeWidth || this.diameter / 10;
+  }
+
+  set strokeWidth(value: number) {
+    this._strokeWidth = coerceNumberProperty(value);
+  }
+
+  /** The radius of the spinner, adjusted for stroke width. */
+  get _circleRadius() {
```

Comment:
Can we turn these private getters into functions? Functions generate less code when transpiled to es5.